raft/src/test/java/org/apache/kafka/raft/RaftClientTestContext.java

```diff
@@ -108,6 +108,17 @@
 
     private final List<RaftResponse.Outbound> sentResponses = new ArrayList<>();
 
+    public static void advanceHighWatermark(RaftClientTestContext context,
```

Comment:
This method will be called in both `KafkaRaftClientSnapshotTest` and `SnapshotWriterTest`, so mark it as public

Reply:
How about making it a object/instance method by removing the static keyword so that the user can use it as follow:
```
context.advanceLeaderHighWatermarkToEndOffset()
```
Note that I renamed the method to include leader and removed all of the parameters. I could be wrong but `RaftClientTestContext` should have all of the information it needs to implement this. Also note, that if you want to generalize this then majority of the nodes need to send a `FETCH` request to the leader for the leader to advance the high-watermark.